x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/AsyncActionStep.java

```diff
@@ -25,7 +26,7 @@ public boolean indexSurvives() {
         return true;
     }
 
-    public abstract void performAction(IndexMetaData indexMetaData, Listener listener);
+    public abstract void performAction(IndexMetaData indexMetaData, ClusterState currentClusterState, Listener listener);
```

Comment:
should we just have `Index` and `ClusterState` and fetch `indexMetaData` internally every time?